pypesto/result/optimize.py

```diff
@@ -269,12 +269,12 @@ def append(
         """
         current_ids = set(self.id)
         if isinstance(optimize_result, OptimizeResult):
-            new_ids = [
+            new_ids = {
                 prefix + identifier
                 for identifier in optimize_result.id
                 if identifier is not None
-            ]
-            if current_ids.isdisjoint(new_ids) and new_ids:
+            }
+            if not current_ids.isdisjoint(new_ids) and new_ids:
```

Comment:
```suggestion
            if current_ids.intersection(new_ids):
```

Reply:
The former is faster. We don't need the full intersection, it's enough to know that the intersection is non-empty.